payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -51,18 +60,100 @@ internal abstract class BaseAddPaymentMethodFragment(
+                    "None of the requested payment methods" +
+                        " (${config.stripeIntent.paymentMethodTypes})" +
+                        " matches the supported payment types" +
+                        " (${SupportedPaymentMethod.values().toList()})"
+                )
+            )
+            return
+        }
+
+        viewBinding.googlePayDivider.setText(
+            if (paymentMethods.contains(SupportedPaymentMethod.Card) && paymentMethods.size == 1) {
+                R.string.stripe_paymentsheet_or_pay_with_card
+            } else {
+                R.string.stripe_paymentsheet_or_pay_using
+            }
+        )
+
+        if (paymentMethods.size > 1) {
+            setupRecyclerView(viewBinding, paymentMethods)
+        }
+
+        replacePaymentMethodFragment(paymentMethods[0])
 
         eventReporter.onShowNewPaymentOptionForm()
     }
 
-    private fun addPaymentMethodFragment() {
+    private fun setupRecyclerView(
+        viewBinding: FragmentPaymentsheetAddPaymentMethodBinding,
+        paymentMethods: List<SupportedPaymentMethod>
+    ) {
+        viewBinding.paymentMethodsRecycler.isVisible = true
+        val layoutManager = object : LinearLayoutManager(
+            activity,
+            HORIZONTAL,
+            false
+        ) {
+            var canScroll = true
+
+            override fun canScrollHorizontally(): Boolean {
+                return canScroll && super.canScrollHorizontally()
+            }
+        }.also {
+            viewBinding.paymentMethodsRecycler.layoutManager = it
+        }
+
+        val adapter = AddPaymentMethodsAdapter(paymentMethods, ::onPaymentMethodSelected).also {
+            viewBinding.paymentMethodsRecycler.adapter = it
+        }
+
+        sheetViewModel.processing.observe(viewLifecycleOwner) { isProcessing ->
+            adapter.isEnabled = !isProcessing
+            layoutManager.canScroll = !isProcessing
+        }
+    }
+
+    @VisibleForTesting
+    internal fun onPaymentMethodSelected(paymentMethod: SupportedPaymentMethod) {
+        replacePaymentMethodFragment(paymentMethod)
+    }
+
+    private fun replacePaymentMethodFragment(paymentMethod: SupportedPaymentMethod) {
         childFragmentManager.commit {
-            add(
+            setCustomAnimations(
+                AnimationConstants.FADE_IN,
+                AnimationConstants.FADE_OUT,
+                AnimationConstants.FADE_IN,
+                AnimationConstants.FADE_OUT
+            )
+            replace(
                 R.id.payment_method_fragment_container,
-                AddCardFragment::class.java,
+                fragmentForPaymentMethod(paymentMethod),
                 arguments
             )
         }
     }
+
+    companion object {
+        private fun fragmentForPaymentMethod(paymentMethod: SupportedPaymentMethod) =
+            when (paymentMethod) {
+                SupportedPaymentMethod.Card -> CardDataCollectionFragment::class.java
```

Comment:
ooc what factors did you weigh in deciding to put this switch here vs. just having the associated fragment defined in the SupportPaymentMethod?